File: src/main/java/catwalks/block/property/UPropertyObject.java
package catwalks.block.property;

/**
 * Created by TheCodeWarrior
 */
public class UPropertyObject<T extends Comparable<T>> extends UPropertyHelper<T>{
	
	public UPropertyObject(String name, Class<T> clazz) {
		super(name, clazz);
	}
	
	@Override
	public boolean isValid(T value) {
		return true;
	}
	
	@Override
	public String valueToString(T value) {
		return "complexData";
	}
}

File: src/main/java/catwalks/block/property/UPropertyObject.kt
package catwalks.block.property

/**
 * Created by TheCodeWarrior
 */
class UPropertyObject<T : Comparable<T>>(name: String, clazz: Class<T>) : UPropertyHelper<T>(name, clazz) {

    override fun isValid(value: T): Boolean {
        return true
    }

    override fun valueToString(value: T): String {
        return "complexData"
    }
}
